src/subaction_animations.jl

```diff
@@ -67,13 +65,61 @@ function disappear(s::Symbol)
 end
 
 function _disappear(video, action, subaction, rel_frame, symbol::Val{:fade_line_width})
-    # t is between 0 and 1
-    t = (rel_frame - first(get_frames(subaction)) + 1)/length(get_frames(subaction))
+    t = get_interpolation(subaction, rel_frame)
     action.current_setting.mul_line_width = 1-t
 end
 
 function _disappear(video, action, subaction, rel_frame, symbol::Val{:fade})
-    # t is between 0 and 1
-    t = (rel_frame - first(get_frames(subaction)) + 1)/length(get_frames(subaction))
+    t = get_interpolation(subaction, rel_frame)
     action.current_setting.mul_opacity = 1-t
 end
+
+"""
+    translate()
+
+Translate a function defined inside an [`Action`](@ref) using an Animation defined
+with Animations.jl.
+
+If you're used to working with Animations.jl this should feel quite natural.
+Instead of defining each movement in its own subaction it's possible to define it in one
+by using an Animation.
+
+# Example
+```
+using Javis, Animations
+
+function ground(args...)
+    background("black")
+    sethue("white")
+end
+
+video = Video(500, 500)
+circle_anim = Animation(
+    [0.0, 0.3, 0.6, 1.0], # must go from 0 to 1
+    # the circle will move from the origin to `Point(150, 0)` then `Point(150, 150)`
+    # and back to the origin `O`.
+    [O, Point(150, 0), Point(150, 150), O],
+    [sineio(), polyin(5), expin(8)],
+)
+javis(
+    video, [
+        BackgroundAction(1:150, ground),
+        Action((args...)->circle(O, 25, :fill); subactions=[
+            SubAction(1:150, circle_anim, translate())
+        ])
+    ], pathname="moving_a_circle.gif"
+)
```

Comment:
One thing that I am somewhat nervous about is the fact that there is two different syntax types we offer: `Javis.jl` Action syntax and `Animations.jl` Animation syntax. I think the way you could fix this is to show both versions of the syntax. Here is the syntax for Javis Actions that achieves the same animation:

```julia
javis(                                                                                 
    video,                                                                             
    [                                                                                  
        BackgroundAction(1:150, ground),                                               
        Action((args...)->circle(O, 25, :fill); subactions = [                         
            SubAction(1:50, sineio(), Translation(150, 0)),                            
            SubAction(51:100, polyin(2), Translation(0, 150)),                         
            SubAction(101:150, expin(8), Translation(-150, -150))                      
        ])                                                                             
    ],                                                                                 
    pathname = "moving_a_circle_javis.gif",                                            
)                              
```

Reply:
Why does that make you nervous? A lot of stuff can be done in different ways :smile: But yeah I can add that to the docstring

Reply:
Thanks! Feel free to resolve this once you are done with adding the docstring. I guess my nervousness has to do with the fact that we do not control Animations.jl, therefore, the syntax could change at any moment and would leave us scrambling. We control Javis syntax so I have no concerns there.